File: src/main/java/com/dv/telegram/command/GetSetting.java
package com.dv.telegram.command;

import com.dv.telegram.WikiBot;
import com.dv.telegram.config.BotSetting;

public class GetSetting extends BasicBotCommand {

    @Override
    public String getName() {
        return GetSetting.class.getSimpleName();
    }

    @Override
    public String getDescription() {
        return String.format("%s <settingName> — получить значение настройки с названием <settingName>.", getCommandText());
    }

    @Override
    public boolean useMarkdownInResponse() {
        return true;
    }

    @Override
    public String getDefaultCommandName() {
        return "/getSetting";
    }

    @Override
    public String getResponse(String text, WikiBot bot) {
        int commandStartIndex = text.indexOf(getCommandText());
        if (commandStartIndex < 0) {
            return unknownSettingResponse();
        }

        int commandEndIndex = commandStartIndex + getCommandText().length();
        if (commandEndIndex >= text.length()) {
            return unknownSettingResponse();
        }

        String settingName = text.substring(commandEndIndex).trim();

        BotSetting<?> botSetting = bot.getSettings().getBotSetting(settingName);
        if (botSetting == null) {
            return unknownSettingResponse();
        }

        String value = BasicBotCommand.getSettingValueForMarkdown(botSetting);
        return String.format("*%s*%n%s", botSetting.getName(), value);
    }

    private String unknownSettingResponse() {
        return "Неизвестное имя настройки.";
    }
}

File: src/main/kotlin/com/dv/telegram/command/GetSetting.kt
package com.dv.telegram.command

import com.dv.telegram.WikiBot
import org.telegram.telegrambots.meta.api.objects.Update

class GetSetting : BasicBotCommand() {
    override val name: String = javaClass.simpleName

    override fun getDescription(bot: WikiBot) =
        "`${bot.botName} $commandText <settingName>` — получить значение настройки с названием `<settingName>`."

    override fun useMarkdownInResponse() = true

    override val defaultCommandName = "/getSetting"

    override fun getResponse(text: String, bot: WikiBot, update: Update): String {
        val commandStartIndex = text.indexOf(commandText)
        if (commandStartIndex < 0) {
            return unknownSettingResponse()
        }

        val commandEndIndex = commandStartIndex + commandText.length
        if (commandEndIndex >= text.length) {
            return unknownSettingResponse()
        }

        val settingName = text.substring(commandEndIndex).trim()

        val botSetting = bot.settings.getBotSetting(settingName)
            ?: return unknownSettingResponse()

        val value = getSettingValueForMarkdown(botSetting)
        return "*${botSetting.name}*\n$value"
    }

    private fun unknownSettingResponse() = "Неизвестное имя настройки."
}
